# Foundations of Data Science
Winter 2025
Group 17

## Team members
- Jack (Xianguo) Hao
- Kunwar Dhruv Sethi
- Eric Ramalheiro
- Lucas Hele
- Faiza Qayoom
- Miguel Morales 222

**Import Modules**

In [22]:
import pandas as pd

**Read data**

In [ ]:

df = pd.read_csv('../data/cost_of_living_us.csv', na_values = '\xa0')

**Inspect Data** 

- Check for missing values and Garbage values

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31430 entries, 0 to 31429
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   case_id                 31430 non-null  int64  
 1   state                   31430 non-null  object 
 2   isMetro                 31430 non-null  bool   
 3   areaname                31430 non-null  object 
 4   county                  31430 non-null  object 
 5   family_member_count     31430 non-null  object 
 6   housing_cost            31430 non-null  float64
 7   food_cost               31430 non-null  float64
 8   transportation_cost     31430 non-null  float64
 9   healthcare_cost         31430 non-null  float64
 10  other_necessities_cost  31430 non-null  float64
 11  childcare_cost          31430 non-null  float64
 12  taxes                   31430 non-null  float64
 13  total_cost              31430 non-null  float64
 14  median_family_income    31420 non-null

In [25]:
## Describe for int / float values
df.describe()

case_id  housing_cost     food_cost  transportation_cost  \
count  31430.000000  31430.000000  31430.000000         31430.000000   
mean    1589.311804  11073.673539   8287.504557         13593.856381   
std      917.218414   4165.606147   3271.140249          1640.456562   
min        1.000000   4209.311280   2220.276840          2216.461440   
25%      792.000000   8580.000000   5801.424360         12535.159800   
50%     1593.000000  10416.000000   8129.156280         13698.164400   
75%     2386.000000  12444.000000  10703.624280         14765.758500   
max     3171.000000  61735.587600  31178.619600         19816.482000   

       healthcare_cost  other_necessities_cost  childcare_cost         taxes  \
count     31430.000000            31430.000000    31430.000000  31430.000000   
mean      13394.031748             7015.318377     9879.584233   7657.714782   
std        5204.545710             2397.415490     6778.223399   3339.795571   
min        3476.379960             2611.642080        0.000000   1027.800756   
25%        9667.440000             5286.354120     5341.621590   5597.970360   
50%       13082.700000             6733.056120    10166.340120   6898.468860   
75%       16657.816800             8413.090230    14276.377800   8790.207270   
max       37252.274400            28829.443200    48831.085200  47753.390400   

          total_cost  median_family_income  
count   31430.000000          31420.000000  
mean    70901.683601          68315.997017  
std     21846.545235          16886.970245  
min     30087.662400          25529.976562  
25%     53776.019400          57223.988281  
50%     70977.682800          65955.605469  
75%     85371.341100          76136.070312  
max    223717.548000         177662.468750

In [26]:
#Data describe for Object values
df.describe(include=['O'])

state                                           areaname  \
count   31430                                              31430   
unique     51                                               2561   
top        TX  Atlanta-Sandy Springs-Roswell, GA HUD Metro FM...   
freq     2540                                                240   

                   county family_member_count  
count               31430               31430  
unique               1877                  10  
top     Washington County                1p0c  
freq                  300                3143

check for diplicate rows

In [27]:
duplicate_rows_df = df[df.duplicated()]
duplicate_rows_df.count()

case_id                   0
state                     0
isMetro                   0
areaname                  0
county                    0
family_member_count       0
housing_cost              0
food_cost                 0
transportation_cost       0
healthcare_cost           0
other_necessities_cost    0
childcare_cost            0
taxes                     0
total_cost                0
median_family_income      0
dtype: int64

Observations:

1. The data set contains 31430 rows and 15 columns 
2. Max case_id value is 3171, however each row should have unique value and max value should be 31430 does this mean there are duplicate case_id values, and if yes what does that mean ? Additionally, as per duplicate check there are no duplicate rows. Further analysis is required and also if it would make sense to type cast this column to string / object.
3. Childcare_cost has $0 values - does that mean there is missing values ?
4. There are 51 unique state values - further investigation is required, considering there are 50 States in United States
5. Does values in 'areaname' columns are concatenated ? for instance 'Atlanta-Sandy.. ' and 'GA HUD Metro ..' is same area name or separate areanames
6. values in family_member_count also requires further investigation as it seems values in this field are coded. For instance does 1p0c means 1 person and no child ? if yes, this column is good candidate for Feature Engineer i.e. split parent and childrens information. 